File: src/main/java/com/hedvig/backoffice/services/payments/dto/Transaction.java
package com.hedvig.backoffice.services.payments.dto;

import java.time.Instant;
import java.util.UUID;
import javax.money.MonetaryAmount;
import lombok.Value;
import org.javamoney.moneta.Money;

@Value
public class Transaction {
  UUID id;
  MonetaryAmount amount;
  Instant timestamp;
  String type;
  String status;

  public static Transaction fromTransactionDTO(UUID id, TransactionDTO dto) {
    return new Transaction(
        id,
        Money.of(dto.getAmount(), dto.getCurrency()),
        dto.getTimestamp(),
        dto.getTransactionType(),
        dto.getTransactionStatus());
  }
}

File: src/main/java/com/hedvig/backoffice/services/payments/dto/Transaction.kt
package com.hedvig.backoffice.services.payments.dto

import java.time.Instant
import java.util.*
import javax.money.MonetaryAmount

data class Transaction(
  var id: UUID,
  var amount: MonetaryAmount,
  var timestamp: Instant,
  var type: String,
  var status: String
) {
  companion object {
    fun fromTransactionDTO(id: UUID, dto: TransactionDTO) = Transaction(
      id = id,
      amount = dto.amount,
      timestamp = dto.timestamp,
      type = dto.transactionType,
      status = dto.transactionStatus)
  }
}
